apps/cowswap-frontend/src/modules/usdAmount/apis/getCoingeckoUsdPrice.ts

```diff
@@ -70,6 +70,6 @@ export async function getCoingeckoUsdPrice(currency: Token): Promise<number | nu
       return Promise.reject(error)
     })
     .then((res: CoinGeckoUsdQuote) => {
-      return res[Object.keys(res)[0]]?.usd || null
+      return res[Object.keys(res)[0]]?.usd?.toString() || null
```

Comment:
Let's take this query as an example https://api.coingecko.com/api/v3/simple/token_price/ethereum?contract_addresses=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48&vs_currencies=usd

The result:

```
{
  "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48": {
    "usd": 1
  }
}
```

See the [docs](https://www.coingecko.com/en/api/documentation)

Since we are filtering by a single address, the result will include at most 1 key, which is the address of the token we are looking for.

That's why it's safe to get the first item from the list of keys.

Of course, we could instead use the token address we queried.
Have to check how it works for the native token, though

Reply:
Updated to use the token address instead